api/v2/views/base.py

```diff
@@ -24,9 +24,16 @@ def unresolved_requests_only(fn):
     @wraps(fn)
     def wrapper(self, request, *args, **kwargs):
```

Comment:
Looks like this wrapper function I initially created is getting complicated.  It really was only meant to check for requests that were in a non final state. Ideally, any permission level concerns should be handler elsewhere. Just thought I would share my thoughts.